pkg/kubelet/client/kubelet_client.go

```diff
@@ -155,10 +157,12 @@ type NodeConnectionInfoGetter struct {
 	insecureSkipTLSVerifyTransport http.RoundTripper
 	// preferredAddressTypes specifies the preferred order to use to find a node address
 	preferredAddressTypes []v1.NodeAddressType
+
+	clusterTransportGetter func(string) (*url.URL, http.RoundTripper, error)
 }
 
 // NewNodeConnectionInfoGetter creates a new NodeConnectionInfoGetter.
-func NewNodeConnectionInfoGetter(nodes NodeGetter, config KubeletClientConfig) (ConnectionInfoGetter, error) {
+func NewNodeConnectionInfoGetter(nodes NodeGetter, config KubeletClientConfig, fleet *registry.FleetClientSet) (ConnectionInfoGetter, error) {
```

Comment:
Is this handling the proxy of Node resources?

Reply:
GetConnectionInfo depend on requestinfo in  ctx